### DISCUSSION TITLE
getting ERC721InvalidReceiver revert on mintNft function in BasicNft contract

### DISCUSSION THREAD

**Discussion by NandanNNN** [Cyfrin/foundry-full-course-cu#1268] [Q&A]
After deploying NFT contract on etherscan getting below error
Deployed contract --> 0x394cbE9FC534f73116E0c0cb8060376D93Ae360a ---> https://sepolia.etherscan.io/address/0x394cbe9fc534f73116e0c0cb8060376d93ae360a#code

├─ [139900] BasicNft::mintNft(ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json%22)
│ ├─ emit Transfer(from: 0x0000000000000000000000000000000000000000, to: MintBasicNft: [0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519], tokenId: 0)
│ ├─ [170] MintBasicNft::onERC721Received(MintBasicNft: [0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519], 0x0000000000000000000000000000000000000000, 0, 0x)
│ │ └─ ← "EvmError: Revert"
│ └─ ← "ERC721InvalidReceiver(0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519)"
└─ ← "ERC721InvalidReceiver(0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519)"

**Comment by xyz899** [comment]
Hi,

I think your error is due to the fact that the contract is using Openzeppelin ERC721 I's and L's and some override solidity functions are missing ? "ERC721InvalidReceiver" suggests that the contract you're trying to interact with does not implement the ERC721 interface correctly. Check if you have have all the necessary functions and if you can't use OZ Wizard Contracts.

**Reply by NandanNNN** [reply]
@Winter9998 thanks for responding. can you please elaborate what you mean by" I's and L's and some override solidity functions are missing ?"  and what is OZ wizard contracts?

**Reply by xyz899** [reply]
> @Winter9998 thanks for responding. can you please elaborate what you mean by" I's and L's and some override solidity functions are missing ?"  and what is OZ wizard contracts?


Interfaces and Libraries. OZ = OpenZeppelin Contracts. Check the contracts on Github themselves for more infos. There are functions that needs to be overridden in order to use the ERC721 Implementation

**Reply by NandanNNN** [reply]
I found the issue. Actually i didn't broadcasted mintNFT function in interactions.s.sol .

Code i made change

'''
function mintNftOnContract(address contractAddress) public {
vm.startBroadcast(); //this i didn't put earlier
BasicNft(contractAddress).mintNft(PUG);
vm.stopBroadcast();
}
'''

**Reply by xyz899** [reply]
> I found the issue. Actually i didn't broadcasted mintNFT function in interactions.s.sol .
> 
> 
> 
> Code i made change
> 
> 
> 
> '''
> 
> function mintNftOnContract(address contractAddress) public {
> 
> vm.startBroadcast(); //this i didn't put earlier
> 
> BasicNft(contractAddress).mintNft(PUG);
> 
> vm.stopBroadcast();
> 
> }
> 
> '''

Problem is you didn't share any code so we're stuck just trying to understand with the foundry CLI only
